open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/VRPSubmissionPayloadValidator.java

```diff
@@ -0,0 +1,512 @@
+                        .get(ConsentExtensionConstants.REMITTANCE_INFO);
+                Object remittanceInformationInitiation = initiationParameterOfConsentInitiation
+                        .get(ConsentExtensionConstants.REMITTANCE_INFO);
+
+                if (remittanceInformationSubmission instanceof JSONObject &&
+                        remittanceInformationInitiation instanceof JSONObject) {
+                    JSONObject remittanceInformationSub = (JSONObject) initiationOfSubmission
+                            .get(ConsentExtensionConstants.REMITTANCE_INFO);
+                    JSONObject remittanceInformationInit = (JSONObject) initiationParameterOfConsentInitiation
+                            .get(ConsentExtensionConstants.REMITTANCE_INFO);
+
+                    validationResult = VRPSubmissionPayloadValidator.validateRemittanceInfo
+                            (remittanceInformationSub, remittanceInformationInit);
+                    if (!Boolean.parseBoolean(validationResult.
+                            getAsString(ConsentExtensionConstants.IS_VALID_PAYLOAD))) {
+                        return validationResult;
+                    }
+                } else {
+                    return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_INVALID,
+                            ErrorConstants.INITIATION_REMITTANCE_INFO_NOT_JSON_ERROR);
+                }
+
+            } else {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                        ErrorConstants.INITIATION_REMITTANCE_INFO_PARAMETER_NOT_FOUND);
+            }
+        }
+
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
+        return validationResult;
+    }
+
+    /**
+     * Validates the instruction between submission and initiation JSONObjects.
+     *
+     * @param submission The submission JSONObject from submission request.
+     * @param initiation The initiation JSONObject from initiation request, here we consider the initiation parameter
+     *                   since the creditor account from the initiation request need to be retrieved.
+     * @return A JSONObject indicating the validation result. It contains a boolean value under the key
+     * ConsentExtensionConstants.IS_VALID_PAYLOAD, indicating whether the payload is valid. If the
+     * validation fails, it returns a JSONObject containing error details with keys defined in ErrorConstants.
+     */
+    public static JSONObject validateInstruction(JSONObject submission,
+                                                 JSONObject initiation) {
+
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, false);
+        validationResult.put(ConsentExtensionConstants.ERROR_CODE, "");
+        validationResult.put(ConsentExtensionConstants.ERROR_MESSAGE, "");
+
+        if (submission != null && initiation != null) {
```

Comment:
what happens `submission ` and `initiation ` both null?

Reply:
Fixed in [a613632]